src/java/org/apache/cassandra/hints/HintsStore.java

```diff
@@ -236,6 +236,17 @@ void markDispatchOffset(HintsDescriptor descriptor, InputPosition inputPosition)
         dispatchPositions.put(descriptor, inputPosition);
     }
 
+    // returns total length of all files belonging to the hints store, in bytes.
+    long getTotalFileSize()
+    {
+        long total = 0;
+        for (HintsDescriptor descriptor : Iterables.concat(dispatchDequeue, corruptedFiles))
+        {
+            total += descriptor.file(hintsDirectory).length();
+        }
+        return total;
```

Comment:
we _could_ use streams here :)

```java
return Stream.concat(dispatchDequeue.stream(), corruptedFiles.stream())
               .mapToLong(descriptor -> descriptor.file(hintsDirectory).length())
               .sum();
```

Reply:
Yep. We could. I'd keep the current as it produces (negligible) less objects.